File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        if not nums:
            return 0
        
        nums.sort()
        
        prev = nums[0]
        #print(prev)
        licz = 1
        maks = 0
        for i in range (1, len(nums)):
            #print(nums[i])
            if nums[i] - nums[i-1] == 1:
                #print("here: ", nums[i], " , ", nums[i-1] )
                licz += 1
            elif nums[i] != nums[i-1]:
                #print("here2: ", nums[i], " , ", nums[i-1] )
                maks = max(maks, licz)
                licz = 1

        maks = max(maks, licz)
        return maks
            
            

        
        

File: Data Structures & Algorithms/house-robber/submission-3.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        if(nums.size() == 0)return 0;
        if(nums.size() ==1) return nums[0];
        int old=0;
        int prev =nums[0];
        
        for(int i=1; i < nums.size(); i++)
        {
            int temp = max(old+nums[i], prev);
            old = prev;
            prev = temp;
        } 
        return prev;
    }
};
